extensions/applicationinsights-clickanalytics-js/src/Interfaces/Datamodel.ts

```diff
@@ -0,0 +1,283 @@
+/**
+ * @copyright Microsoft 2020
+ */
+
+import { IEventTelemetry } from "@microsoft/applicationinsights-common";
+
+
+/**
+ * ClickAnalytics Configuration
+ */
+export interface IClickAnalyticsConfiguration {
+    /**
+     * Automatic capture configuration. Default is true
+     */
+    autoCapture?: boolean;
+    /**
+     * Callbacks configuration
+     */
+    callback?: IValueCallback;
+    /**
+     * Page tags 
+     */
+    pageTags?: { [name: string]: string | number | boolean | string[] | number[] | boolean[] | object };
+    /**
+     * Core data configuration
+     */
+    coreData?: ICoreData;
+    /**
+     * Custom Data Tags provided to ovverride default tags used to capture click data.
+     */
+    dataTags?: ICustomDataTags;
+    /**
+     * Enables the logging of values after a "#" character of the URL. Default is "false."
+     */
+    urlCollectHash?: boolean;
+    /**
+     * Enables the logging of the query string of the URL. Default is "false."
+     */
+    urlCollectQuery?: boolean;
+    /**
+     * Validator to use for the data-bhvr value
+     */
+    behaviorValidator?: (value: string | number) => string | number;
```

Comment:
Would the input ever be a number? If you are getting it from the attribute, I would think it's always a string

Reply:
the _getBehavior() function used in capturePageAction takes in datastructure to override behavior. There is a possibility that users might pass in numbers.